test/VarianceTest.cc

```diff
@@ -0,0 +1,353 @@
+/*
+ * Copyright(c) 2019 Netflix, Inc.
+ * SPDX - License - Identifier: BSD - 2 - Clause - Patent
+ */
+
+/******************************************************************************
+ * @file InvTxfm2dAsmTest.c
+ *
+ * @brief Unit test for variance, mse, sum square functions:
+ * - aom_variance{4-128}x{4-128}_{c,sse2,avx2}
+ * - aom_get_mb_ss_sse2
+ * - aom_mse16x16_{c,avx2}
+ *
+ * @author Cidana-Ryan
+ *
+ ******************************************************************************/
+#include <math.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <new>
+// workaround to eliminate the compiling warning on linux
+// The macro will conflict with definition in gtest.h
+#ifdef __USE_GNU
+#undef __USE_GNU  // defined in EbThreads.h
+#endif
+#ifdef _GNU_SOURCE
+#undef _GNU_SOURCE  // defined in EbThreads.h
+#endif
+#include "aom_dsp_rtcd.h"
+#include "random.h"
+#include "util.h"
+#include "gtest/gtest.h"
+
+using svt_av1_test_tool::SVTRandom;  // to generate the random
+namespace {
+#define MAX_BLOCK_SIZE (128 * 128)
+
+static uint32_t variance_ref(const uint8_t *src, const uint8_t *ref, int width,
```

Comment:
The function should be renamed to mse_ref

Reply:
it for both mse and variance, might not change its name.